src/coprocessor/dag/aggr_fn/impl_avg.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use cop_datatype::{EvalType, FieldTypeFlag, FieldTypeTp};
+use tipb::expression::{Expr, ExprType, FieldType};
+
+use super::summable::Summable;
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::codec::mysql::Tz;
+use crate::coprocessor::dag::expr::EvalContext;
+use crate::coprocessor::dag::rpn_expr::{RpnExpression, RpnExpressionBuilder};
+use crate::coprocessor::{Error, Result};
+
+/// The parser for AVG aggregate function.
+pub struct AggrFnDefinitionParserAvg;
+
+impl super::parser::Parser for AggrFnDefinitionParserAvg {
+    fn check_supported(&self, aggr_def: &Expr) -> Result<()> {
+        use cop_datatype::FieldTypeAccessor;
+        use std::convert::TryFrom;
+
+        assert_eq!(aggr_def.get_tp(), ExprType::Avg);
+        if aggr_def.get_children().len() != 1 {
+            return Err(box_err!(
+                "Expect 1 parameter, but got {}",
+                aggr_def.get_children().len()
+            ));
+        }
+
+        // Check whether or not the children's field type is supported. Currently we only support
+        // Double and Decimal and does not support other types (which need casting).
```

Comment:
I think we could support all types here since tidb won't push down casting functions for each kind of data.

Reply:
Yes of course. That's why the RPN expression is outputted by the aggregation function instead of the framework, which allows the function to insert the CAST in the future.